com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -882,6 +891,57 @@ internal void InitializeSensors()
 #endif
         }
 
+        void InitializeActuators()
+        {
+            ActuatorComponent[] attachedActuators;
+            if (m_PolicyFactory.UseChildActuators)
+            {
+                attachedActuators = GetComponentsInChildren<ActuatorComponent>();
+            }
+            else
+            {
+                attachedActuators = GetComponents<ActuatorComponent>();
+            }
+
+            // Support legacy OnActionReceived
+            var param = m_PolicyFactory.BrainParameters;
+            switch (param.VectorActionSpaceType)
+            {
+                case SpaceType.Continuous:
+                    vectorActuator = new ContinuousVectorActuator(param.VectorActionSize);
+                    break;
+                case SpaceType.Discrete:
+                    vectorActuator = new DiscreteVectorActuator(param.VectorActionSize);
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException($"Unknown SpaceType used:{param.VectorActionSpaceType}");
+            }
+
+            if (vectorActuator != null)
+            {
+                actuators = new ActuatorList(attachedActuators.Length);
+                actuators.Add(vectorActuator);
+            }
+
+            foreach (var actuatorComponent in attachedActuators)
```

Comment:
I think we need a check (either around here or in ActuatorList.Add) to make sure all the space types are the same.

Reply:
yes, that's a TODO item.